framework/src/utils/FunctionParserUtils.C

```diff
@@ -191,22 +191,34 @@ template <>
 void
 FunctionParserUtils<false>::functionsOptimize(SymFunctionPtr & parsed_function)
 {
+  // set desired epsilon for optimization!
+  auto tmp_eps = parsed_function->epsilon();
+  parsed_function->setEpsilon(_epsilon);
```

Comment:
The epsilon documentation string doesn't mention anything about optimization; the documentation string just makes it seem like there's a single epsilon. But this temporary setting of epsilon makes it seem like there's one epsilon for optimization and another epsilon elsewhere. If there are two epsilons at play let's make sure it's indicated in the documentation string, and should there be user ability to set both?

Reply:
There is one global epsilon variable. I could add the functionality of setting different epsilons for both stages.